### PR TITLE
winetricks: Adaptation for NTFS (filesystem)

### PR DESCRIPTION
**Pull request by Kreyren**
**DO NOT MERGE** - Investigation needed

Partially fixes: https://github.com/Winetricks/winetricks/issues/1363

Signed-off-by: Jacob Hrbek <kreyren@rixotstudio.cz>

### CODE REVIEW

File: src/winetricks [hunk 1]
```diff
@@ -21732,6 +21756,19 @@ execute_command()
...
+## Instructions to make a proper NTFS setup (https://jp-andre.pagesperso-orange.fr/permissions.html)
+[ -n "$WINETRICKS_IGNORE_NTFS" ] && case "$W_PLATFORM" in nix_*)
+    case "$(mount | grep "type ntfs")" in "/dev/??? on / type ntfs*")
+      w_warn "Winetricks detected Windows NT filesystem which is not reliable to be used on nix \
+- https://www.howtogeek.com/fyi/microsoft-explains-why-windows-10s-october-2018-update-was-deleting-peoples-files/ \
+- https://github.com/Winetricks/winetricks/issues/1363 \
+Make sure that your NTFS filesystem on root is configured correctly (https://jp-andre.pagesperso-orange.fr/permissions.html)"
+    esac
+esac
+
```
Comment (Kreyren) [inline src/winetricks]: Adapted based on discussion in #winehq

**Comment by Kreyren** [comment]
```
possible bashism in src/winetricks line 410 ($UID should be "$(id -ru)"):

          "$([ $(stat -c "%U") = 'root' ] && [ -n "$UID" ] && printf '%s\n' "$WINETRICKS_SUDO")" chmod +r "$filename" || w_die "Unable to set readability on '$filename'"

possible bashism in src/winetricks line 421 ($UID should be "$(id -ru)"):

          "$([ $(stat -c "%U") = 'root' ] && [ -n "$UID" ] && printf '%s\n' "$WINETRICKS_SUDO")" chmod +r "$filename" || w_die "Unable to set readability on '$filename'"

possible bashism in src/winetricks line 439 ($UID should be "$(id -ru)"):

        "$([ $(stat -c "%U") = 'root' ] && [ -n "$UID" ] && printf '%s\n' "$WINETRICKS_SUDO")" chmod +r "$2" || w_die "Unable to set readability on '$2'"

possible bashism in src/winetricks line 449 ($UID should be "$(id -ru)"):

          "$([ $(stat -c "%U") = 'root' ] && [ -n "$UID" ] && printf '%s\n' "$WINETRICKS_SUDO")" chmod +r "$2" || w_die "Unable to set readability on '$2'"
```

Using `$UID` -> i don't think that relying on `/usr/bin/id` is better then using /bin/bash enviromental variable

**Comment by eli-schwartz** [comment]
- `id -u` is already used in several places within the winetricks script.
- `id -u` is a required POSIX functionality: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/id.html
- the script explicitly uses /bin/sh, not the bash interpreter, presumably because why yes, it is supposed to be able to run on any POSIX shell, like dash (which it is extremely fair to assume Debian systems tend to do)
- checkbashisms warned you due to the fact that it was a problem, so why would you automatically think it is wrong?

**Comment by Kreyren** [comment]
@eli-schwartz I was assuming that `id` is not present on cygwin or on system that doesn't use coreutils, but that doesn't seem to be a thing. 

thanks for info, logic adapted

**Comment by Kreyren** [comment]
Is there a way to detect file ownership on POSIX?

**Comment by Kreyren** [comment]
Invalid: 

```
[user@host ~]$ cd Downloads/
[user@host Downloads]$ wget http://ix.io/686
--2019-10-07 21:26:58--  http://ix.io/686
Resolving ix.io (ix.io)... 66.172.11.73
Connecting to ix.io (ix.io)|66.172.11.73|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10 [text/plain]
Saving to: ‘686’

686                 100%[===================>]      10  --.-KB/s    in 0s      

2019-10-07 21:26:59 (333 KB/s) - ‘686’ saved [10/10]

[user@host Downloads]$ file 686 
686: ASCII text
[user@host Downloads]$ cat 686
something
-rw-r--r-- 1 user user 10 Oct  7 21:26 Downloads/686
```

**Pull request closed without merging**